### PR TITLE
avoid exceptions for empty lists

### LINKED ISSUE #37: Unable to read empty collections
Steps to reproduce issue:
1. create table with frozen collection column
    ```
    CREATE TABLE test_frozen_collection (
        key int,
        collection frozen<list<text>>,
        PRIMARY KEY (key)
    );
    ```
1. insert row with empty collection
    ```
    INSERT INTO test_frozen_collection (key, collection) VALUES (1, []);
    ```
1. attempt to read row in pyspark
    ```
    import pyspark_cassandra
    pyspark_cassandra.monkey_patch_sc(sc)
    rdd = sc.cassandraTable('test_keyspace', 'test_frozen_collection')
    rdd.where('key=?', 1).collect()
    ```
1.
    error:
    `net.razorvine.pickle.PickleException: couldn't introspect javabean: java.lang.IllegalArgumentException: wrong number of arguments`

Note, if the collection column contains a `null` value, `pyspark` is happy
```
INSERT INTO test_frozen_collection (key, collection) VALUES (1, null);
```

### PR DESCRIPTION
**Pull request by ndamclean**
- convert scala lists to java lists when pickling
- update pyrolite to latest version

fixes #37

### CODE REVIEW

**Comment by ndamclean** [comment]
If this looks good to you, want to do a 0.9.1 release with this?

**Comment by anguenot** [comment]
Hey @ndamclean, thank you for the fix! Possible to get some tests along with these changes?

File: src/main/scala/pyspark_util/Pickling.scala [hunk 1]
```diff
@@ -101,6 +102,7 @@ class Pickling extends Serializable {
     Pickler.registerCustomPickler(classOf[Set.Set4[_]], ListPickler)
     Pickler.registerCustomPickler(classOf[HashSet1[_]], ListPickler)
     Pickler.registerCustomPickler(classOf[HashTrieSet[_]], ListPickler)
+    Pickler.registerCustomPickler(classOf[Map[_, _]], MapPickler)
     Pickler.registerCustomPickler(classOf[WithDefault[_, _]], MapPickler)
```
Comment (ndamclean) [inline src/main/scala/pyspark_util/Pickling.scala:106]: @anguenot When I added unit tests, I found a similar issue with empty maps.

Registering a custom pickler for the `Map` type seems to fix this, but I confess I am not familiar with Scala and don't know what is going on here with the `WithDefault` and `MapN` types.

Comment (ndamclean) [reply src/main/scala/pyspark_util/Pickling.scala:106]: Looking at Scala docs... it looks like these are subclasses of `Map`. Is it necessary to register the picklers for all the subtypes?
```
    Pickler.registerCustomPickler(classOf[WithDefault[_, _]], MapPickler)
    Pickler.registerCustomPickler(classOf[Map1[_, _]], MapPickler)
    Pickler.registerCustomPickler(classOf[Map2[_, _]], MapPickler)
    Pickler.registerCustomPickler(classOf[Map3[_, _]], MapPickler)
    Pickler.registerCustomPickler(classOf[Map4[_, _]], MapPickler)
```
I tried removing these lines and the tests are passing for me locally, but I'm not sure that there are not potential problems not covered by the test suite.

Comment (anguenot) [reply src/main/scala/pyspark_util/Pickling.scala:106]: @ndamclean  wanna open a  new issue for that one so we can trace potential issues separately?

**Comment by anguenot** [comment]
Thanks @ndamclean

File: src/main/scala/pyspark_util/Pickling.scala (hunk 1 above)
Comment (ndamclean) [reply src/main/scala/pyspark_util/Pickling.scala:106]: I looked at the pyrolite code and concluded it's probably best to have all the subclasses registered separately for the sake of efficiency. 
```
	protected IObjectPickler getCustomPickler(Type t)
	{
		IObjectPickler pickler;
		if(customPicklers.TryGetValue(t, out pickler))
			return pickler;		// exact match
		
		// check if there's a custom pickler registered for an interface or abstract base class
		// that this object implements or inherits from.
		foreach(var x in customPicklers) {
			if(x.Key.IsAssignableFrom(t)) {
				return x.Value;
			}
		}


		return null;
	}
```
https://github.com/irmen/Pyrolite/blob/cd05ec1ddbf96b8b40925ba68b57dc7db0c46ef0/dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/Pickler.cs#L313-L328

**Pull request merged**